### PR TITLE
Fix overlong impl

### LINKED ISSUE #1048: Overlong impl
```
impl<BorrowType, K, V, NodeType, HandleType> Handle<NodeRef<BorrowType, K, V, NodeType>, HandleType> {
```

I assume this is an off by n error.

### PR DESCRIPTION
**Pull request by dawirstejeck**
Fix #1048.

Will also look into splitting the impl at the 'for'.

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -470,7 +471,7 @@ pub fn format_impl(context: &RewriteContext, item: &ast::Item, offset: Indent) -
             result.push_str(" for ");
         }
 
-        let budget = try_opt!(context.config.max_width.checked_sub(result.len()));
+        let budget = try_opt!(context.config.max_width.checked_sub(result.len() + 1));
```
Comment (nrc) [inline src/items.rs]: Could you add a comment for why we need the `+ 1` please?

File: tests/source/issue-1048.rs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Test issue-1048
```
Comment (nrc) [inline tests/source/issue-1048.rs]: Could probably just put this test in one of the existing test files.

**Comment by nrc** [comment]
Looks good, just a couple of minor comments, then r+

**Comment by dawirstejeck** [comment]
@nrc I think one of the checks timed out. Could you please restart them?

**Comment by nrc** [comment]
Thank you for the PR!

**Comment by dawirstejeck** [comment]
My pleasure!

**Pull request merged**